src/tokenizer.java

```diff
@@ -21,152 +21,115 @@
-                else if (";".equals(tok.toString())) {
-                    tokenList.add(new end_stmt(lineCount));
-                }
-                else if ("Integer".equals(tok.toString())) {
-                    tokenList.add(new int_label(lineCount));
-                }
-                else if ("print".equals(tok.toString())) {
-                    tokenList.add(new print_label(lineCount));
-                }
-                else if ("(".equals(tok.toString())) {
-                    tokenList.add(new start_paren(lineCount));
-                }
-                else if ("String".equals(tok.toString())) {
-                    tokenList.add(new str_label(lineCount));
-                }
-                else if ("*".equals(tok.toString()) || "/".equals(tok.toString()) || "^".equals(tok.toString()) ||
-                        "+".equals(tok.toString()) || "-".equals(tok.toString())) {
-                    tokenList.add(new op(lineCount, tok.toString()));
-                }
-                else if ("\"".equals(tok.toString())) {
-                    //String literal
-                    tokenList.add(new quote(lineCount));
-
-                    tok = new StringBuilder();
-
                     i++;
                     while (i < lineLength) {
-                        char ch = line.charAt(i);
-                        if (ch == '"')
+                        ch = line.charAt(i);
+                        if (ch == '"') {
+                            tok.append(ch);
                             break;
-
-                        if (!((ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') || (ch >= '0' && ch <= '9') || ch == ' '))
+                        }
+                        if (!Character.isAlphabetic(ch) && !Character.isDigit(ch) && ch != ' ')
                             errorPrinter.throwError(lineCount, new Syntax("Illegal character '" + ch +
                                     "' detected in string literal"));
-
                         tok.append(ch);
-
                         i++;
                     }
-
                     if (i == lineLength)
                         errorPrinter.throwError(lineCount, new Syntax("Strings cannot wrap lines"));
-
                     tokenList.add(new str_token(lineCount, tok.toString()));
-                    tokenList.add(new quote(lineCount));
-                }
-                else if (tok.charAt(0) == '+' || tok.charAt(0) == '-' || tok.charAt(0) == '.' || Character.isDigit(tok.charAt(0))) {
+                } else if (Character.isDigit(ch) || ch == '.') { // Integer or Double
                     boolean isDouble = false;
-                    int intV = -1;
-                    double dblV = -1;
-                    try {
-                        intV = Integer.parseInt(tok.toString());
-                    } catch (NumberFormatException n) {
-                        try {
-                            dblV = Double.parseDouble(tok.toString());
+                    int z = -1;
```

Comment:
What exactly is `z`? Would prefer a more appropriate name for this var.

Reply:
In mathematics, Z denotes the set of integers, hence the var z

Reply:
why not something like `intIn`?